pkg/services/grafana-apiserver/service.go

```diff
@@ -248,7 +250,23 @@ func (s *service) start(ctx context.Context) error {
 		openapi.GetOpenAPIDefinitionsWithoutDisabledFeatures(defsGetter),
 		openapinamer.NewDefinitionNamer(Scheme, scheme.Scheme))
 
+	// Add the custom routes to service discovery
+	serverConfig.OpenAPIV3Config.PostProcessSpec3 = getOpenAPIPostProcessor(builders)
+
 	serverConfig.SkipOpenAPIInstallation = false
+	serverConfig.BuildHandlerChainFunc = func(delegateHandler http.Handler, c *genericapiserver.Config) http.Handler {
```

Comment:
is this to register routes? did the `NonGoRestfulMux` not work for this use case?

Reply:
we tried that before and were unable to get it to work when used *with* resources -- but there may be a way

Reply:
@toddtreece this is used because the order in which route processing happens between NonGoRestfulMux and REST routes is that NonGoRestfulMux always hits after.

Since we were adding non traditional routes for custom handlers, we were getting 404s from K8s standard REST resource handling and never hitting NonGoRestfulMux.

By overriding the handler chain itself using BuildHandlerChainFunc, we were finally able to do the ordering of our custom handlers first. Hence we stuck to this solution.

Ref: [director](https://github.com/kubernetes/kubernetes/blob/master/staging/src/k8s.io/apiserver/pkg/server/handler.go#L116) is the pertinent piece which decides the ordering between REST and nonGoRestfulMux.